src/main/java/TaskList.java

```diff
@@ -0,0 +1,158 @@
+     */
+    protected void addMessage() {
+        System.out.println(" Got it. I've added this task: ");
+        System.out.println(taskList.get(taskList.size() - 1));
+        if (taskList.size() == 1) {
+            System.out.println("Now you have 1 task in your list.");
+        } else {
+            System.out.println("Now you have " + taskList.size() + " tasks in the list.");
+        }
+    }
+
+    /**
+     *Marks the task as done upon user's request and prints the message upon successful completion.
+     *
+     * @param x the position of the task in the list.
+     */
+    protected void markAsDone(int x) {
+        taskList.get(x).markAsDone();
+        System.out.println("Nice! I've marked this task as done:");
+        System.out.println(taskList.get(x));
+    }
+
+    /**
+     *Deletes the task in the list upon user's request and prints the number of tasks remaining upon deletion.
+     *
+     * @param y the position of the task in the list
+     */
+    protected void deleteTask(int y) {
+        if (taskList.size() == 0) {
+            System.out.println("The task list is empty");
+        } else {
+            System.out.println("Noted. I've removed this task:");
+            System.out.println(taskList.get(y - 1));
+            taskList.remove(y - 1);
+            if (taskList.size() == 1) {
+                System.out.println("Now you have 1 task in your list.");
+            } else {
+                System.out.println("Now you have " + taskList.size() + " tasks in the list.");
+            }
+        }
+
+    }
+
+    /**
+     *Finds and prints all the tasks containing the word requested by the user.
+     *
+     * @param s The word that the user wants to find.
+     */
+    protected void find(String s) {
+        int count = 0;
+        System.out.println("Here are the matching tasks in your list:");
+        for (Task t : taskList) {
+            if (t.description.contains(s)) {
+                System.out.println(t);
+                count += 1;
+            }
+        }
+        if (count == 0) {
+            System.out.println("There are no matching tasks in your list");
+        }
+    }
+
+    /**
+     *Prints all the tasks in the list.
+     */
+    protected void getList() {
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 1; i <= taskList.size(); i += 1) {
+            System.out.println(i + ". " + taskList.get(i - 1));
+        }
+    }
+
+    /**
+     *Reads and processes the event given by the user.
+     * Stores the information in the list after processing.
+     *
+     * @param b The information of the event being processed.
+     * @throws ParseException If the data is not in the required format i.e. MM/dd/yyyy HH:mm.
+     */
+    protected void readEvent(String b) throws ParseException {
+
+        if (b.length() == 0) {
+            System.out.println(" OOPS!!! the description of a event cannot be empty. ");
```

Comment:
Use of DukeException here will be better to print exceptions related to the application.